### ISSUE TITLE
`consider-using-with` false positive with AsyncExitStack

### ISSUE THREAD

**Issue by marcinsulikowski** [PyCQA/pylint#5337]
### Bug description

The following code:
```py
# pylint: disable=missing-docstring

import contextlib

async def bug() -> None:
    async with contextlib.AsyncExitStack() as stack:
        stack.enter_context(open("/tmp/x", "wb"))
```
triggers a false-positive:
```
$ pylint a.py 
************* Module a
a.py:7:28: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
```
https://github.com/PyCQA/pylint/pull/4665 was supposed to fix that but for some reason the fix doesn't work for `AsyncExitStack` (it works with `ExitStack`).

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:7:28: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
```

### Expected behavior

There should be no `consider-using-with` for context managers passed to `AsyncExitStack.enter_context`.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.3
Python 3.9.9 (main, Nov 16 2021, 03:05:18) 
[GCC 7.5.0]
```

### OS / Environment

GNU/Linux (Ubuntu)

**Comment by DudeNr33** [comment]
Thank you for reporting this. I will look into it.

**Comment by DudeNr33** [comment]
There are two problems:
1. the code currently only checks for ``enter_context`` and needs to be extended to also check ``enter_async_context``. This is the easier part. https://github.com/PyCQA/pylint/blob/2e7b2c20cb08e739dfcc83c8ae1508a96ee8e04d/pylint/checkers/refactoring/refactoring_checker.py#L140-L154
2. ``utils.safe_infer(node.parent.func)`` returns ``Uninferable``. This requires an astroid update.

**Comment by marcinsulikowski-es** [comment]
> There are two problems:
>
> 1. the code currently only checks for `enter_context` and needs to be extended to also check `enter_async_context`. This is the easier part

Note that my example uses `enter_context`, the non-async one. `open` returns a regular (non-async) context manager so it must not be passed to `enter_async_context`. There's really just one problem here – the astroid issue which makes `pylint` unable to tell that `enter_context` is used.

**Comment by DudeNr33** [comment]
Oh, you are absolutely right! 
(Note to self: take a close look at Github Copilot's suggestions...)

However, ``enter_async_context`` should probably also be taken into account by ``consider-using-with``.
